broadinstitute/ddp-study-server
# DSP-7581: adding data dictionary to export and parse refactor

This includes some work based on https://github.com/broadinstitute/ddp-study-server/pull/2095 and responding to the comments there.  This is opened as a fresh PR since it's now different enough from what's on the RC that it counts as a fresh feature.  this also includes work in response to comments on https://github.com/broadinstitute/ddp-study-server/pull/2121

This adds data dictionary support and also cleans up yet more potential null pointer edge cases. The data dictionary is a complete listing of all study variables, useful for researchers doing work on the data set. Pressing "download" will now download a .zip file of both the exported data, and a data dictionary excel document. See the file below for an example data dictionary generated off of development AT

[DataDictionary.xlsx](https://github.com/broadinstitute/ddp-study-server/files/9271277/DataDictionary.xlsx)

TO TEST:

load a study participant list
select "download"
download with the default options
observe that a zip file is downloaded, containing both a PArticipant data export, and a "DataDictionary.xslx" file
